### PR TITLE
Shipping Label Payments: disable editing for non-store owners

### LINKED ISSUE #3597: Shipping Labels Payments: disable editing for non-store owners
If the user creating the shipping label is not the store owner, they can't change payment method, so we need to display the below screen to let them know that a payment method is already in file and that they can't edit it:

Screen Shot 2021-02-18 at 12 53 04

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes #3597, disable editing payments for non-store owners, and display a warning about it.

#### Testing

1. Make sure your WCS setup has a payment (or multiple) methods.
2. Create an administrator account in your store.
3. Login to the app using this account.
4. Create an order that satisfies the shipping label creation.
5. Open the order details.
6. Click on "create shipping label"
7. Pass all the steps until you reach the payment details.
8. Click on edit button in the payments step.
9. Confirm that editing the payment methods is disabled, and there is a warning to explain it.
10. Confirm that you can edit the email receipts checkbox.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by anitaa1990** [comment]
Moving this PR to 6.2, since we're doing a code freeze of 6.1. Feel free to ping me if this PR should be included in 6.1

**Comment by nbradbury** [comment]
@hichamboushaba There's a minor merge conflict on this.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelPaymentViewModel.kt [hunk 1]
```diff
@@ -49,6 +49,7 @@ class EditShippingLabelPaymentViewModel @AssistedInject constructor(
                     PaymentMethodUiModel(paymentMethod = it, isSelected = it.id == accountSettings.selectedPaymentId)
                 },
                 canManagePayments = accountSettings.canManagePayments,
+                canEditSettings = accountSettings.canEditSettings || accountSettings.canManagePayments,
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelPaymentViewModel.kt:54]: @hichamboushaba I found this line confusing at first. Perhaps add a comment explaining why the model's `canEditSettings` is set on either `accountSettings.canEditSettings` or  `accountSettings.canManagePayments`?

**Comment by nbradbury** [comment]
Code looks and works great! Just had a couple of nit picks.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelPaymentViewModel.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelPaymentViewModel.kt:54]: Done in 9a8c1f8e095acb3d0b38d969cbfd85ebb36baa9f

**Comment by hichamboushaba** [comment]
Thanks @nbradbury, ready for another round 🙂

**Review by nbradbury — APPROVED** [review]
Looks good!

**Pull request merged**